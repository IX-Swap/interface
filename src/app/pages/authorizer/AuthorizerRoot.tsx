import React from 'react'
import { AuthorizerRouter } from 'app/pages/authorizer/router/AuthorizerRouter'

export const AuthorizerRoot: React.FC = () => {
  return <AuthorizerRouter />
}
