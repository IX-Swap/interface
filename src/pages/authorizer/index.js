// @flow
import React, { Suspense } from 'react';
import { withRouter, Route, Link, RouteProps } from 'react-router-dom';
import { Grid, Box, Typography } from '@material-ui/core';
import PageTitle from 'components/PageTitle';

const Banks = React.lazy(() => import('./banks'));
const Deposits = React.lazy(() => import('./deposits'));
const Withdrawals = React.lazy(() => import('./withdrawals'));
const IndividualIdentities = React.lazy(() =>
  import('./individual-identities')
);
const CorporateIdentities = React.lazy(() => import('./corporate-identities'));

const routes = [
  {
    route: '/authorizer/banks',
    title: 'Banks',
    component: Banks,
  },
  {
    route: '/authorizer/deposits',
    title: 'Deposits',
    component: Deposits,
  },
  {
    route: '/authorizer/withdrawals',
    title: 'Withdrawals',
    component: Withdrawals,
  },
  {
    route: '/authorizer/individual-identities',
    title: 'Individual Identities',
    component: IndividualIdentities,
  },
  {
    route: '/authorizer/corporate-identities',
    title: 'Corporate Identities',
    component: CorporateIdentities,
  },
];

const Links = () => (
  <>
    {routes.map((route) => (
      <Link key={route.title} to={route.route}>
        {route.title}
      </Link>
    ))}
  </>
);

const Routes = () => (
  <Suspense fallback={<span>loading</span>}>
    {routes.map((route) => (
      <Route key={route.title} path={route.route} component={route.component} />
    ))}
  </Suspense>
);

const getTitle = (path: string): string => {
  switch (path) {
    case '/authorizer/banks':
      return 'Banks';
    case '/authorizer/deposits':
      return 'Deposits';
    case '/authorizer/withdrawals':
      return 'Withdrawals';
    case '/authorizer/individual-identities':
      return 'Individual Identities';
    case '/authorizer/corporate-identities':
      return 'Corporate Identities';
    default:
      return '';
  }
};

function Authorizer(props: RouteProps) {
  const { location } = props;

  return (
    <>
      <Links />
      <br />
      <Grid container title="Accounts" justify="center" alignItems="center">
        <Grid item xs={12}>
          <Box my={4}>
            {location && <PageTitle title={getTitle(location.pathname)} />}
          </Box>
        </Grid>
        <Routes />
      </Grid>
    </>
  );
}

export default withRouter(Authorizer);
