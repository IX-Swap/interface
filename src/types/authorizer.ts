import { AuthorizableStatus } from './util'
import { DataroomFile } from './dataroomFile'
import { AppFeature } from 'types/app'
import {
  CorporateIdentity,
  IndividualIdentity
} from 'app/pages/identity/types/forms'

export interface AuthorizationInfo {
  authorizer: string
  comment: string
  sharedWithUser: boolean
  timestamp: string
}

export interface AuthorizationInfoWithStatus extends AuthorizationInfo {
  _id: string
  status: AuthorizableStatus
}

export interface AuthorizationOverride {
  _id: string
  authorizer: string
  withdrawalAddress?: string
  releaseDate?: string
  timestamp: string
}

export interface Authorizable {
  _id: string
  createdAt: string
  updatedAt: string
  status: AuthorizableStatus
  authorizations: AuthorizationInfoWithStatus[]
  authorizationDocuments?: DataroomFile[]
  authorization?: AuthorizationInfo
  level?: string
  assignedAt?: string
}

export interface AuthorizableWithIdentity extends Authorizable {
  identity?: {
    individual: IndividualIdentity
    corporates: CorporateIdentity[]
  }
}

export const DataroomFeature = {
  [AppFeature.BankAccounts]: 'authorization/accounts/bank-accounts',
  [AppFeature.CashDeposits]: 'authorization/accounts/deposits',
  [AppFeature.CashWithdrawals]: 'authorization/accounts/withdrawals',
  [AppFeature.DigitalSecurityWithdrawals]:
    'authorization/accounts/security-withdrawals',
  [AppFeature.WithdrawalAddresses]:
    'authorization/accounts/withdrawal-addresses',
  [AppFeature.Individuals]: 'authorization/identity/individuals',
  [AppFeature.Corporates]: 'authorization/identity/corporates',
  [AppFeature.Offerings]: 'authorization/issuance/dsos',
  [AppFeature.Commitments]: 'authorization/issuance/commitments'
}

export interface AuthorizerViewParams {
  category: keyof typeof DataroomFeature
  itemId: string
  ownerId: string
}

export interface Authorization {
  authorizer?: string
}
