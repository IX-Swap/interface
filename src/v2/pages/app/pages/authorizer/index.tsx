import React from 'react'
import {
  Switch,
  Route,
  useRouteMatch,
  Redirect
} from 'react-router-dom'

import routes from './routes'

const Routes = ({ parent }: { parent: string }) => (
  <Switch>
    <Route
      exact
      path={parent + '/'}
      component={() => <Redirect to={parent + '/banks'} />}
    />
    {routes.map(({ path, component, exact }) => (
      <Route
        key={path}
        path={`${parent}${path}`}
        component={component}
        exact={exact}
      />
    ))}
  </Switch>
)

const MemoedRoutes = React.memo(Routes)

const Authorizer = () => {
  const match = useRouteMatch()

  return (
    <>
      <MemoedRoutes parent={match.path} />
    </>
  )
}

export default Authorizer
