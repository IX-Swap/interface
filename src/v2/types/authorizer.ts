import { AuthorizableStatus } from './util'
import { DataroomFile } from './dataroomFile'
import { CorporateIdentity, IndividualIdentity } from './identity'

export interface AuthorizationInfo {
  authorizer: string
  comment: string
  sharedWithUser: boolean
  timestamp: string
}

export interface AuthorizationInfoWithStatus extends AuthorizationInfo {
  _id: string
  status: AuthorizableStatus
}

export interface Authorizable {
  _id: string
  createdAt: string
  updatedAt: string
  status: AuthorizableStatus
  authorizations: AuthorizationInfoWithStatus[]
  authorizationDocuments?: DataroomFile[]
  authorization?: AuthorizationInfo
  level?: string
}

export interface AuthorizableWithIdentity extends Authorizable {
  identity?: {
    individual: IndividualIdentity
    corporates: CorporateIdentity[]
  }
}

export enum DataroomFeature {
  banks = 'authorization/accounts/bank-accounts',
  deposits = 'authorization/accounts/deposits',
  withdrawals = 'authorization/accounts/withdrawals',
  dsWithdrawals = 'authorization/accounts/security-withdrawals',
  individualIdentities = 'authorization/identity/individuals',
  corporateIdentities = 'authorization/identity/corporates',
  offerings = 'authorization/issuance/dsos',
  commitments = 'authorization/issuance/commitments'
}

export interface AuthorizerViewParams {
  category: keyof typeof DataroomFeature
  itemId: string
  ownerId: string
}
