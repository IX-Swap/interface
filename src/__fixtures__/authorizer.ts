import { Bank } from 'types/bank'
import { APIServiceResponse } from 'services/api/types'
import { CashDeposit } from 'types/cashDeposit'
import { Asset } from 'types/asset'
import { CorporateIdentity, IndividualIdentity } from 'types/identity'
import { user } from '__fixtures__/user'
import { CashWithdrawal } from 'types/cashWithdrawal'
import { DSWithdrawal } from 'types/dsWithdrawal'
import { DigitalSecurityOffering, DSOInsight } from 'types/dso'
import { AuthorizationInfo } from 'types/authorizer'
import { Commitment } from 'types/commitment'
import { emptyFile } from '__fixtures__/file'
import { network } from './network'
import { withdrawalAddress } from './withdrawalAddress'

export const asset: Asset = {
  _id: '5f732c538a568b50914d8372',
  deleted: false,
  createdBy: '5f73240373d4ab4b15fc1b2e',
  symbol: 'SGD',
  name: 'Singapore Dollars',
  type: 'Currency',
  numberFormat: {
    currency: 'SGD'
  },
  createdAt: '2020-09-29T12:45:07.411Z',
  updatedAt: '2020-10-05T03:14:17.244Z',
  description: 'test asset description'
}

export const address = {
  city: 'Omsk',
  country: 'Russian Federation',
  state: 'Siberia',
  line1: 'Address line 1',
  line2: 'Address line 2',
  postalCode: '123456'
}

export const authorizationInfo: AuthorizationInfo = {
  authorizer: 'id',
  comment: '',
  sharedWithUser: false,
  timestamp: ''
}

export const corporate: CorporateIdentity = {
  _id: '1',

  logo: '',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  documents: [],
  declarations: [],
  status: 'Submitted',
  beneficialOwners: [],
  companyAddress: address,
  companyLegalName: 'InvestaX',
  countryOfFormation: 'Singapore',
  dateOfIncorporation: '01-01-2000',
  directors: [],
  registrationNumber: '123456',
  representatives: [],
  email: '',
  contactNumber: '',
  user,
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: []
}

export const individual: IndividualIdentity = {
  photo: '',
  _id: '1',
  email: 'email@example.com',
  annualIncome: '100000',
  bankAccountName: '',
  bankAccountNumber: '',
  bankName: '',
  contactNumber: '1234567890',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  countryOfResidence: 'Russian Federation',
  dob: 'DOB',
  employer: 'InvestaX',
  employmentStatus: 'Employed',
  firstName: 'John',
  lastName: 'Doe',
  gender: 'M',
  houseHoldIncome: '100000',
  industryOfEmployment: 'IT',
  maritalStatus: 'Married',
  middleName: '',
  nationality: 'Russian',
  occupation: 'Occupied',
  // politicallyExposed: false,
  sourceOfWealth: '___',
  status: 'Submitted',
  authorization: authorizationInfo,
  authorizationDocuments: [],
  authorizations: [],
  declarations: [],
  documents: [],
  address,
  user
}

export const dsWithdrawal: DSWithdrawal = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  user: 'user',
  amount: 100000,
  hold: 'hold',
  level: '1',
  memo: 'memo',
  status: 'Approved',
  recipientWallet: '0000000',
  asset,
  transaction: '',
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: [],
  identity: {
    individual,
    corporates: []
  }
}

export const dsoInsight: DSOInsight = {
  activityCount: 65,
  approvedcommitmentCount: 1,
  collectedOn: '2020-11-24T17:06:23.555Z',
  commitmentCount: 0,
  commitmentTotal: 0,
  investorCount: 1,
  raisedMax: 400,
  raisedMin: 400,
  raisedTotal: 400
}

export const dso: DigitalSecurityOffering = {
  _id: '1',
  asset: 'asset',
  businessModel: 'business model',
  capitalStructure: 'capital structure',
  network: network._id,
  corporate,
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  createdBy: 'created by',
  currency: asset,
  deleted: false,
  deploymentInfo: undefined,
  distributionFrequency: 'distribution frequency',
  dividendYield: 1,
  documents: [],
  equityMultiple: 0,
  fundraisingMilestone: 'fundraising milestone',
  grossIRR: 0,
  interestRate: 1,
  introduction: 'introduction',
  investmentPeriod: 1,
  investmentStructure: 'investment structure',
  issuerName: 'issuer name',
  launchDate: '01-01-2000',
  leverage: 0,
  logo: '5f4f7d87f3e2c40bbab8a3f1',
  minimumInvestment: 100,
  pricePerUnit: 1,
  subscriptionDocument: emptyFile,
  team: [],
  tokenName: 'token name',
  tokenSymbol: 'token symbol',
  totalFundraisingAmount: 100000,
  useOfProceeds: 'use of proceeds',
  policyBuilder: undefined,
  user: user._id,
  authorizationDocuments: [],
  authorization: authorizationInfo,
  authorizations: [],
  identity: {
    individual,
    corporates: []
  },
  status: 'Approved',
  insight: dsoInsight
}

export const commitment: Commitment = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  status: 'Approved',
  currency: asset,
  createdBy: 'user',
  dso,
  identity: {
    individual,
    corporates: [corporate]
  },
  totalAmount: 0,
  authorizationDocuments: [],
  authorizationOverrides: [],
  authorizationOverride: undefined,
  authorization: undefined,
  authorizations: [],
  hold: 'dsadas',
  level: undefined,
  numberOfUnits: 0,
  pricePerUnit: 0,
  signedSubscriptionDocument: emptyFile,
  user,
  withdrawalAddress: withdrawalAddress
}

export const bank: Bank = {
  _id: '1',
  status: 'Approved',
  deleted: false,
  bankName: 'Rocketbank',
  bankAccountNumber: '1234567890',
  accountHolderName: 'Oleg Tinkoff',
  swiftCode: 'SWIFTCODE',
  authorized: true,
  authorizationDocuments: [],
  authorizations: [],
  supportingDocuments: [],
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  authorization: authorizationInfo,
  identity: {
    corporates: [corporate],
    individual
  },
  user,
  address,
  currency: asset
}

export const cashDeposit: CashDeposit = {
  _id: '1',
  amount: 10000,
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  depositCode: 'DEPOSIT',
  status: 'Approved',
  user: '???',
  bankAccount: bank,
  individual,
  corporates: [corporate],
  asset,
  identity: {
    individual,
    corporates: [corporate]
  },
  authorizations: [],
  authorizationDocuments: [],
  authorization: undefined
}

export const cashWithdrawal: CashWithdrawal = {
  _id: '1',
  createdAt: '01-01-2000',
  updatedAt: '01-01-2000',
  user: 'user',
  amount: 100000,
  corporates: [],
  hold: 'hold',
  level: '1',
  memo: 'memo',
  status: 'Approved',
  bank,
  individual,
  asset,
  identity: {
    individual,
    corporates: []
  },
  authorizations: [],
  authorizationDocuments: [],
  authorization: undefined
}

export const authorizerURLs = {
  approve: `/${bank._id}/approve`,
  reject: `/${bank._id}/reject`
}

export const approveResponseSuccess: APIServiceResponse = {
  message: 'Approved successfully',
  success: true
}

export const approveResponseFailure: APIServiceResponse = {
  message: 'Could not approve',
  success: false
}

export const rejectResponseSuccess: APIServiceResponse = {
  message: 'Rejected successfully',
  success: true
}

export const rejectResponseFailure: APIServiceResponse = {
  message: 'Could not reject',
  success: false
}
