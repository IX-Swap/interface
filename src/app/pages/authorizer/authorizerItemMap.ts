import { AppFeature, AuthorizerCategory } from 'types/app'
import { BankPreview } from 'app/components/BankPreview/BankPreview'
import { urlParams } from 'config/appURL'
import { DSWithdrawalPreview } from 'app/components/DSWithdrawalPreview/DSWithdrawalPreview'
import { WithdrawalPreview } from 'app/components/WithdrawalPreview/WithdrawalPreview'
import { DepositView } from 'app/components/DepositView/DepositView'
import { WithdrawalAddressPreview } from 'app/components/WithdrawalAddressPreview/WithdrawalAddressPreview'
import { CommitmentPreview } from 'app/components/CommitmentPreview/CommitmentPreview'
import { DSOView } from 'app/components/DSO/DSOView'
import { CorporateView } from 'app/pages/identity/components/CorporateView'
import { IndividualView } from 'app/pages/identity/components/IndividualView'
import { AuthorizerRoute } from 'app/pages/authorizer/router'

export interface AuthorizerMapItem {
  uri: string
  component: any
  paramKey: string
  title: string
  listRoute: keyof typeof AuthorizerRoute
}

export const authorizerItemMap: Record<
  AuthorizerCategory,
  AuthorizerMapItem
> = {
  [AppFeature.BankAccounts]: {
    uri: '/accounts/banks/list',
    component: BankPreview,
    paramKey: urlParams.itemId,
    title: 'Bank Account',
    listRoute: 'banks' as const
  },
  [AppFeature.Offerings]: {
    uri: '/issuance/dso/list',
    component: DSOView,
    paramKey: urlParams.itemId,
    title: 'Digital Security Offering',
    listRoute: 'offerings' as const
  },
  [AppFeature.DigitalSecurityWithdrawals]: {
    uri: '/accounts/security/withdrawals',
    component: DSWithdrawalPreview,
    paramKey: urlParams.itemId,
    title: 'Digital Security Withdrawal',
    listRoute: 'dsWithdrawals' as const
  },
  [AppFeature.CashWithdrawals]: {
    uri: '/accounts/cash/withdrawals',
    component: WithdrawalPreview,
    paramKey: urlParams.itemId,
    title: 'Cash Withdrawal',
    listRoute: 'cashWithdrawals' as const
  },
  [AppFeature.Corporates]: {
    uri: '/identity/corporates/list',
    component: CorporateView,
    paramKey: urlParams.itemId,
    title: 'Corporate Identity',
    listRoute: 'corporateIdentities' as const
  },
  [AppFeature.Individuals]: {
    uri: '/identity/individuals/list',
    component: IndividualView,
    paramKey: urlParams.itemId,
    title: 'Individual Identity',
    listRoute: 'individualIdentities' as const
  },
  [AppFeature.CashDeposits]: {
    uri: '/accounts/cash/deposits',
    component: DepositView,
    paramKey: urlParams.itemId,
    title: 'Cash Deposit',
    listRoute: 'cashDeposits' as const
  },
  [AppFeature.Commitments]: {
    uri: '/issuance/commitments/list',
    component: CommitmentPreview,
    paramKey: urlParams.itemId,
    title: 'Commitment',
    listRoute: 'commitments' as const
  },
  [AppFeature.WithdrawalAddresses]: {
    uri: '/accounts/withdrawal-addresses/list',
    component: WithdrawalAddressPreview,
    paramKey: urlParams.itemId,
    title: 'Withdrawal Address',
    listRoute: 'withdrawalAddresses' as const
  }
}
