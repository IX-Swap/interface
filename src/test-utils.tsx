import React from 'react'
import { Router } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'
import {
  createGenerateClassName,
  StylesProvider
} from '@material-ui/core/styles'
import { ThemeProvider } from '@material-ui/styles'
import Themes from './v2/themes'
import { history } from 'v2/history'
import { UserProvider } from 'v2/auth/context'
import { UserStore } from 'v2/auth/context/store'
import { PasswordResetProvider } from 'v2/auth/context/password-reset'
import PasswordResetStore from 'v2/auth/context/password-reset/store'
import { PasswordResetStep } from 'v2/auth/context/password-reset/types'
import { AuthorizerTableStore } from 'v2/app/pages/authorizer/context/store'
import { AuthorizerTableStoreProvider } from 'v2/app/pages/authorizer/context'
import { DepositStore } from 'v2/app/pages/accounts/pages/banks/context/store'
import { DepositStoreProvider } from 'v2/app/pages/accounts/pages/banks/context'
import { ServicesProvider } from 'v2/services/useServices'
import { renderHook, RenderHookResult } from '@testing-library/react-hooks'
import { BreadcrumbsProvider } from 'v2/hooks/useBreadcrumbs'

const generateClassName = createGenerateClassName({
  productionPrefix: 'ix'
})

const BaseProviders: React.FC = ({ children }) => {
  return (
    <StylesProvider generateClassName={generateClassName}>
      <ThemeProvider theme={Themes.default}>
        <BreadcrumbsProvider>
          <Router history={history}>{children}</Router>
        </BreadcrumbsProvider>
      </ThemeProvider>
    </StylesProvider>
  )
}

const customRenderer = (
  ui: any,
  options?: Omit<RenderOptions, 'queries'>
): RenderResult => {
  return render(ui, { wrapper: BaseProviders, ...options })
}

export const testComponentId = 'test-component'

export const fakeUserStore: Partial<UserStore> = {
  activeTab: 0,
  setActiveTab: jest.fn()
}

export const renderWithUserStore = (
  ui: any,
  store?: Partial<UserStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <UserProvider value={{ ...fakeUserStore, ...store }}>
        {children}
      </UserProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakePasswordResetStore: Partial<PasswordResetStore> = {
  currentStep: PasswordResetStep.Request,
  setCurrentStep: jest.fn()
}

export const renderWithPasswordResetStore = (
  ui: any,
  store?: Partial<PasswordResetStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <PasswordResetProvider value={{ ...fakePasswordResetStore, ...store }}>
        {children}
      </PasswordResetProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeAuthorizerTableStore: Partial<AuthorizerTableStore> = {
  idKey: '__',
  uri: '/api/source',
  _getItemId: jest.fn()
}

export const renderWithAuthorizerTableStore = (
  ui: any,
  store?: Partial<AuthorizerTableStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <AuthorizerTableStoreProvider
        value={{ ...fakeAuthorizerTableStore, ...store }}
      >
        {children}
      </AuthorizerTableStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const fakeDepositStore: Partial<DepositStore> = {}

export const renderWithDepositStore = (
  ui: any,
  store?: Partial<DepositStore>
): RenderResult => {
  const WithUserProvider: React.FC = ({ children }) => (
    <BaseProviders>
      <DepositStoreProvider value={{ ...fakeDepositStore, ...store }}>
        {children}
      </DepositStoreProvider>
    </BaseProviders>
  )

  return render(ui, { wrapper: WithUserProvider })
}

export const renderHookWithServiceProvider = (
  hookFn: any,
  store: object = {}
): RenderHookResult<any, any> => {
  const WithServiceProvider: React.FC = ({ children }) => (
    <ServicesProvider value={store}>{children}</ServicesProvider>
  )

  return renderHook(hookFn, { wrapper: WithServiceProvider })
}

export * from '@testing-library/react'
export { customRenderer as render }
