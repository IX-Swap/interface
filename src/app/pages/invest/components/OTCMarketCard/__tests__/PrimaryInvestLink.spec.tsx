import React from 'react'
import { render } from 'test-utils'
import { dso } from '__fixtures__/authorizer'
import {
  PrimaryInvestLink,
  PrimaryInvestLinkProps
} from 'app/pages/invest/components/OTCMarketCard/PrimaryInvestLink'

describe('PrimaryInvestLink', () => {
  const primaryProps: PrimaryInvestLinkProps = {
    type: 'Primary',
    data: dso
  }

  const OTCProps: PrimaryInvestLinkProps = {
    type: 'OTC',
    data: dso
  }
  const TopOffersProps: PrimaryInvestLinkProps = {
    type: 'Primary',
    data: dso
  }

  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('renders Button with correct props when type is Primary', () => {
    const { getByTestId } = render(<PrimaryInvestLink {...primaryProps} />)

    const button = getByTestId('otc-card-link')

    expect(button).toBeInTheDocument()
    expect(button).not.toHaveAttribute('disabled')
    expect(button).toHaveTextContent('Invest')
  })

  it('renders Button with correct props when type is Primary', () => {
    const { getByTestId } = render(<PrimaryInvestLink {...TopOffersProps} />)

    const button = getByTestId('otc-card-link')

    expect(button).toBeInTheDocument()
    expect(button).not.toHaveAttribute('disabled')
    expect(button).toHaveTextContent('Invest')
  })

  it('renders Button with correct props when type is OTC', () => {
    const { getByTestId } = render(<PrimaryInvestLink {...OTCProps} />)

    const button = getByTestId('otc-card-link')

    expect(button).toBeInTheDocument()
    expect(button).toHaveAttribute('disabled')
    expect(button).toHaveTextContent('Trade')
  })
})
