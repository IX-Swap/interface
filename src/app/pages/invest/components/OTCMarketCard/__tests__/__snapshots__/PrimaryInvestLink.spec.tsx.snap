// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimaryInvestLink should match snapshot on button click 1`] = `
<div>
  <button
    class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root Mui-disabled  css-13e45tm-MuiButtonBase-root-MuiButton-root"
    data-testid="otc-card-link"
    disabled=""
    style="font-size: 16px; margin-top: 16px;"
    tabindex="-1"
    type="button"
  >
    Trade
  </button>
</div>
`;
